### PR TITLE
Don't override resources only on Windows OS

### PR DESCRIPTION
**Pull request by matthyx**
Refactored after our discussion.

### CODE REVIEW

**Comment by mryan43** [comment]
There was another merged PR in the same code to address a windows compatibility issue. Could you please rebase this PR on top of origin ? 

The main goal is to improve features on MAC OS right ? If yes, was the fix for Windows of any help for MAC OS ?

File: core/src/main/java/com/github/swissquote/carnotzet/core/Carnotzet.java [hunk 1]
```diff
@@ -116,9 +116,11 @@ public Carnotzet(CarnotzetConfig config) {
 	public List<CarnotzetModule> getModules() {
 		if (modules == null) {
 			modules = resolver.resolve(config.getTopLevelModuleId(), failOnDependencyCycle);
-			if (SystemUtils.IS_OS_LINUX || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
+			if (!SystemUtils.IS_OS_WINDOWS || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
 				resourceManager.extractResources(modules);
-				modules = computeServiceIds(modules);
+			}
+			modules = computeServiceIds(modules);
+			if (!SystemUtils.IS_OS_WINDOWS || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
```
Comment (onigoetz) [inline core/src/main/java/com/github/swissquote/carnotzet/core/Carnotzet.java]: Doing the same check twice seems inefficient, why not store it in a variable instead ? would also clarify intent of the check

Comment (matthyx) [reply core/src/main/java/com/github/swissquote/carnotzet/core/Carnotzet.java]: Which check?

**Comment by matthyx** [comment]
> The main goal is to improve features on MAC OS right ? If yes, was the fix for Windows of any help for MAC OS ?

Actually the fix for Windows broke MacOS... as in general we always want to extract and resolve resources, except on Windows.

**Pull request merged**